app/controllers/cloud_subnet_controller.rb

```diff
@@ -52,17 +52,37 @@ def cloud_subnet_form_fields
     }
   end
 
+  def cloud_subnet_networks_by_ems
+    assert_privileges("cloud_subnet_new")
+    available_networks = []
+    CloudNetwork.where(:ems_id => params[:id]).find_each.select do |network|
```

Comment:
`select` is useless here, you're not filtering by anything 

The `find_each.select` could just become `each`,
or even better,

```
available_networks = CloudNetwork.where(:ems_id => params[:id]).map do |network|
  { 'name' => network.name, 'id' => network.ems_ref }
end